components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,212 @@
+        />
+
+        <label htmlFor="duration">Cooking duration:</label>
+        <input
+          type="text"
+          id="duration"
+          name="duration"
+          defaultValue={isUpdate ? recipeDetails.preparationTime ?? "" : null}
+        />
+
+        <label htmlFor="imgurl">Image Url</label>
+        <input
+          type="url"
+          id="imgurl"
+          name="imgurl"
+          defaultValue={isUpdate ? recipeDetails.imageURL ?? "" : null}
+        />
+
+        <StyledFieldSet>
+          <legend>Ingredients:</legend>
+          <input
+            type="text"
+            name="amount1"
+            placeholder="Amount"
+            required
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[0].quantity ?? "" : null
+            }
+          />
+          <input
+            type="text"
+            name="ingredient1"
+            placeholder="Ingredient"
+            required
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[0].name ?? "" : null
+            }
+          />
+          <input
+            type="text"
+            name="amount2"
+            placeholder="Amount"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[1].quantity ?? "" : null
+            }
+          />
+          <input
+            type="text"
+            name="ingredient2"
+            placeholder="Ingredient"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[1].name ?? "" : null
+            }
+          />
+
+          <input
+            type="text"
+            name="amount3"
+            placeholder="Amount"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[2].quantity ?? "" : null
+            }
+          />
+          <input
+            type="text"
+            name="ingredient3"
+            placeholder="Ingredient"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[2].name ?? "" : null
+            }
+          />
+
+          <input
+            type="text"
+            name="amount4"
+            placeholder="Amount"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[3].quantity ?? "" : null
+            }
+          />
+          <input
+            type="text"
+            name="ingredient4"
+            placeholder="Ingredient"
+            defaultValue={
+              isUpdate ? recipeDetails.ingredients[3].name ?? "" : null
+            }
+          />
+        </StyledFieldSet>
+
+        <label htmlFor="description">Description:</label>
+        <textarea
+          name="description"
+          rows="5"
+          defaultValue={isUpdate ? recipeDetails.description ?? "" : null}
+        />
+
+        <button type="submit">
+          {isUpdate ? "Save Changes" : "Add Recipe"}
+        </button>
+      </StyledForm>
+      <StyledLinkContainer>
+        <StyledLink href="/">Cancel</StyledLink>
```

Comment:
Maybe you want to think about a conditional Link, too. Like this you could handle the cancel button in a way that brings you back to your overview when you come from the create form but to the details page when you come from the update form of a single recipe. It's just an idea though and I'm not completely sure i changed to code correctly for this 😆 
```suggestion
        <StyledLink href={isUpdate ? "recipeDetails.id" : "/"}>Cancel</StyledLink>
```